### ISSUE TITLE
model.val() problem

### ISSUE THREAD

**Issue by Auggie3** [ultralytics/ultralytics#829]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

model.val() different results  after first run (0.49). After .val() creates valid.cache, results are terible. Why? 
![image](https://user-images.githubusercontent.com/87269392/216977506-0b998774-4b53-4bf0-bde0-941c4f44aac0.png)
![image](https://user-images.githubusercontent.com/87269392/216977650-229318cd-e3aa-4d89-a5d9-7e8294962a10.png)

**Comment by Laughing-q** [comment]
@Auggie3 thanks for the reporting! It'd be very helpful if you provide us your command line or script to reproduce the issue. :)

**Comment by Auggie3** [comment]
In  model.train() model is evaluated once at the end which creates
valid.cache. Running model.val() directly after that gives me worse
results. (ignore some outputs, that was me debugging)

уто, 7. феб 2023. у 15:17 Laughing ***@***.***> је написао/ла:

**Comment by Jessmin** [comment]
I met the same problem

**Comment by Auggie3** [comment]
If u delete valid.cache after every validation then it works fine. Is there argument so that valid.cache isnt made after model.val()?

**Comment by Auggie3** [comment]
It seems i didnt send my code before. Here it is:
[UnderwaterTrashDetection (2).ipynb.zip](https://github.com/ultralytics/ultralytics/files/10699728/UnderwaterTrashDetection.2.ipynb.zip)

Every time its the same thing for me.

**Comment by Laughing-q** [comment]
@Auggie3 oh I see it's about the corrupt labels. I've reproduced the issue and made a PR #902  to fix it. :) Thanks!
